### PR TITLE
Support and documentation changes for 2.5.0

### PR DESCRIPTION
**Pull request by lazyfrosch**
This is a general PR to introduce some changes in requirements for 2.5.0

Changes:
- [x] Add FPM config examples
- [x] Add FPM documentation for el6 / el7
- [x] Remove SLES 11 from docs
- [x] Upgrading documentation for 2.5.0
  * Package deprecation
  * Package changes to FPM for some OS
  * Notes for PHP upgrade on SUSE
- [x] Removed RPM spec file from this repo

Also see:
* https://github.com/Icinga/icinga-packaging/pull/60
* https://github.com/Icinga/packages.icinga.com/pull/2

### CODE REVIEW

**Comment by lazyfrosch** [comment]
Also see https://github.com/Icinga/packages.icinga.com/pull/2

File: doc/02-Installation.md [hunk 1]
```diff
@@ -104,17 +98,41 @@ apk update
...
+Please enable it prior installing the Icinga packages.
+ 
 CentOS 7/6:
-```
-yum install epel-release
-```
 
-If you are using RHEL you need enable the `optional` repository and then install
-the [EPEL rpm package](https://fedoraproject.org/wiki/EPEL#How_can_I_use_these_extra_packages.3F).
+    yum install epel-release
```
Comment (dnsmichi) [inline doc/02-Installation.md]: Please keep the format with the three backticks in the docs.

File: library/Icinga/Application/Hook.php [hunk 2]
```diff
@@ -275,21 +277,23 @@ public static function all($name)
...
      */
-    public static function first($name)
+    public static function first($name, $bypassPermission = false)
     {
         $name = self::normalizeHookName($name);
 
         if (self::has($name)) {
             foreach (self::$hooks[$name] as $key => $hook) {
-                if (self::hasPermission($hook)) {
+                if ($bypassPermission || self::hasPermission($hook)) {
```
Comment (dnsmichi) [inline library/Icinga/Application/Hook.php]: That sounds dangerous. Where is that used?

File: doc/02-Installation.md (hunk 1 above)
Comment (lazyfrosch) [reply doc/02-Installation.md]: Is there a reason for that?

File: library/Icinga/Application/Hook.php (hunk 2 above)
Comment (lazyfrosch) [reply library/Icinga/Application/Hook.php]: Within the RequirementHook, reason is that `setup` doesn't use the normal role/permission model

File: doc/02-Installation.md (hunk 1 above)
Comment (dnsmichi) [reply doc/02-Installation.md]: The entire documentation uses that format, and we should keep it that way.

File: library/Icinga/Application/Hook.php (hunk 2 above)
Comment (dnsmichi) [reply library/Icinga/Application/Hook.php]: Ah ok, understood, thanks 👍

**Comment by lazyfrosch** [comment]
@lippserd please merge

**Pull request merged**